src/bp/ui-studio/src/web/views/OneFlow/diagram/index.tsx

```diff
@@ -141,7 +143,18 @@ class Diagram extends Component<Props> {
     this.diagramEngine.registerNodeFactory(new RouterWidgetFactory())
     this.diagramEngine.registerNodeFactory(new ActionWidgetFactory())
     this.diagramEngine.registerNodeFactory(new SuccessWidgetFactory())
-    this.diagramEngine.registerNodeFactory(new TriggerWidgetFactory())
+    this.diagramEngine.registerNodeFactory(
+      new TriggerWidgetFactory({
+        editNodeItem: this.editNodeItem.bind(this),
+        selectedNodeItem: () => this.getStateProperty('editingNodeItem'),
+        deleteSelectedElements: this.deleteSelectedElements.bind(this),
+        getCurrentFlow: () => this.getPropsProperty('currentFlow'),
+        getConditions: () => this.getPropsProperty('conditions'),
+        updateFlowNode: this.updateNodeAndRefresh.bind(this),
+        switchFlowNode: this.switchFlowNode.bind(this),
+        addCondition: this.addCondition.bind(this)
```

Comment:
Can you refactor this to avoid duplication ? There's like 7 elements which are the same for the two types of nodes